src/main/java/com/google/cloud/dataproc/templates/config/OutputConfig.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (C) 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.cloud.dataproc.templates.config;
+
+import com.google.common.base.MoreObjects;
+import com.google.common.collect.ImmutableMap;
+import jakarta.validation.constraints.NotBlank;
+import jakarta.validation.constraints.NotNull;
+import jakarta.validation.constraints.Pattern;
+import java.util.Map;
+
+public class OutputConfig {
+
+  @NotBlank private String format;
+  private String path;
+  @NotNull private Map<String, String> options = ImmutableMap.of();;
+
+  @Pattern(regexp = "Overwrite|ErrorIfExists|Append|Ignore")
```

Comment:
Corresponds to the [SaveMode](https://spark.apache.org/docs/latest/api/java/index.html?org/apache/spark/sql/SaveMode.html) enum